### DISCUSSION TITLE
Question about Vuex and Persisting & Retrieving the State Back into the Store

### DISCUSSION THREAD

**Discussion by kunle-e** [quasarframework/quasar#12512] [General - Components / Directives / etc]
I have a quasar v2.5.5 SPA.
I've tried both vuex-persistedstate and vuex-persist and they both exhibit the same behavior, which is the following:
- the state is persisted to localStorage, but the state is not read back from localStorage to rehydrate the store

I've used vuex-persistedstate outside of quasar years ago and there was nothing much to do but installing the plugin in vuex, and configure any paths. The setting and retrieval of the state into and out of localStorage was handled by the plugin.
But with quasar, after refreshing, the state is not updated to store, but it does remain in localStorage.
If I do something like the below in my state.js, the store is updated from localStorage when the page is refreshed. But this seems weird and wrong because my prior experience was that the plugins handled this internally. So the question is: Is the below the proper way to rehydrate the store from localStorage or have I completely missed something? 

```
// state.js

import { LocalStorage } from 'quasar';
export default function () {
  return {

   // the user object
   user: JSON.parse(LocalStorage.getItem('vuex')).main.user || {},

   userProfile: {
      // the list of competitions created by the app admin
      competitions: JSON.parse(LocalStorage.getItem('vuex')).main.userProfile.competitions || [],
    }
}

**Comment by MilosPaunovic** [comment]
I think you're gonna need to show some more code for someone to be able to efficiently help you. At least the part where you import `vuex-persistedstate`. A minimal repository with reproduction would be the best.

**Reply by kunle-e** [reply]
While setting up a code sandbox using vuex-persistedstate as both a quasar plugin with boot files and as a normal vuex plugin, I saw that that in the boot file createPersistedState was being wrapped by a setTimeout with too much delay. I removed the setTimeout, which was a vestigial artifact from some older examples, and the vuex state was rehydrated on refresh. I also updated to the latest version of vuex (4.0.2).

The problem I do see is that locally, the state updates are not reflected in vue devtools, this may be an issue with vue devtools, but since we can't use vue devtools in a codesanbox, there's not much that can be done to troubleshoot using codesandbox.

***Code sandbox with vuex-persistedstate*** 
https://codesandbox.io/s/boring-nash-1o06xm?file=/src/boot/persist-client.js
